src/client/datascience/jupyter/kernels/kernelDependencyService.ts

```diff
@@ -125,7 +125,7 @@ export class KernelDependencyService implements IKernelDependencyService {
     ): Promise<KernelInterpreterDependencyResponse> {
         // If there's no UI, then cancel installation.
         if (disableUI) {
-            return KernelInterpreterDependencyResponse.cancel;
+            return KernelInterpreterDependencyResponse.uiHidden;
```

Comment:
This way we know when ipykernel isn't installed, and message was not displayed
To handle race conditions, assume user opens a notebook we auto start & ipykenrel isn't installed & skip because ui is disabled, but at the same time user hits the run button.

That's what i want to fix in a seprate PR, this will help.
fixing auto start & per will come later.